receiver/mongodbatlasreceiver/alerts.go

```diff
@@ -72,25 +107,107 @@ func newAlertsReceiver(logger *zap.Logger, cfg AlertConfig, consumer consumer.Lo
 	}
 
 	recv := &alertsReceiver{
-		addr:        cfg.Endpoint,
-		secret:      cfg.Secret,
-		tlsSettings: cfg.TLS,
-		consumer:    consumer,
-		wg:          &sync.WaitGroup{},
-		logger:      logger,
+		addr:          cfg.Endpoint,
+		secret:        cfg.Secret,
+		tlsSettings:   cfg.TLS,
+		consumer:      consumer,
+		mode:          cfg.Mode,
+		projects:      cfg.Projects,
+		retrySettings: baseConfig.RetrySettings,
+		publicKey:     baseConfig.PublicKey,
+		privateKey:    baseConfig.PrivateKey,
+		wg:            &sync.WaitGroup{},
+		pollInterval:  baseConfig.Alerts.PollInterval,
+		maxAlerts:     baseConfig.Alerts.MaxAlertProcessing,
+		doneChan:      make(chan bool, 1),
+		logger:        logger,
+		id:            baseConfig.ID(),
+		storageID:     baseConfig.StorageID,
 	}
 
-	s := &http.Server{
-		TLSConfig: tlsConfig,
-		Handler:   http.HandlerFunc(recv.handleRequest),
+	if recv.mode == alertModePoll {
+		client := internal.NewMongoDBAtlasClient(recv.publicKey, recv.privateKey, recv.retrySettings, recv.logger)
+		recv.client = client
```

Comment:
Is it necessary to allocate `client` here?

Reply:
We could move it to `Start` is that what you are thinking?

Reply:
I realized we can do it in the start but then we'd have to modify the test cases that use the mock client to not use the newMethod(). Seems in line with the current server approach but if you want me to I can make the change back and modify test cases!

Reply:
I just mean, why not `recv.client = internal.NewMongoDB...`?